locations/spiders/sprouts.py

```diff
@@ -0,0 +1,18 @@
+import json
+
+from scrapy.spiders import SitemapSpider
+
+from locations.structured_data_spider import StructuredDataSpider
+
+
+class SproutsFarmersMarket(SitemapSpider, StructuredDataSpider):
+    name = "sprouts_farmer_market"
+    allowed_domains = ["www.sprouts.com"]
+    item_attributes = {"brand": "Sprouts Farmer Market", "brand_wikidata": "Q7581369"}
+    sitemap_urls = ["https://www.sprouts.com/store-sitemap.xml"]
+    sitemap_rules = [(r"^https:\/\/www.sprouts.com\/store\/fl\/*\/*\/", "parse_sd")]
+
+    def post_process_item(self, item, response, ld_data, **kwargs):
+        item["website"] = response.url
+        item["opening_hours"] = ld_data["openingHours"].replace(",", ";")
+        yield item
```

Comment:
The OH are in an invalid format, I've not tested the following across everything so it may need changes.
```suggestion
    def pre_process_data(self, ld_data, **kwargs):
        oh = OpeningHours()

        for days, open_time, open_time_ampm, close_time, close_time_ampm in re.findall(
            r"(\w+-\w+) (\d+:\d+)\s?(AM|PM) (\d+:\d+)\s?(AM|PM)",
            ld_data["openingHours"],
        ):
            start_day, end_day = days.split("-")
            start_day = sanitise_day(start_day)
            end_day = sanitise_day(end_day)
            for day in day_range(start_day, end_day):
                oh.add_range(
                    day,
                    open_time + open_time_ampm,
                    close_time + close_time_ampm,
                    time_format="%I:%M%p",
                )

        ld_data["openingHours"] = oh.as_opening_hours()
```
`website` should be set by `StructuredDataSpider`. I'll fix that in separate PR.

Reply:
Thanks a lot @Cj-Malone
This worked. Below are the 4 sites where time was in lowercase (am,pm) so I added them too and it worked for them.
`Open Daily: 7:00am –10:00pm`
https://www.sprouts.com/store/fl/oviedo/oviedo/
https://www.sprouts.com/store/la/baton-rouge/baton-rouge/
https://www.sprouts.com/store/az/mesa/mesa-signal-butte/
https://www.sprouts.com/store/nj/marlton/marlton/

There is one site where there is no information on whether the time is in AM/PM. So for this opening_hour information is missing.
`Open Daily: 7:00 –10:00`
https://www.sprouts.com/store/az/mesa/mesa-recker/

Reply:
FWIW you could have done it with `re.IGNORECASE`, but this works. Thank you.

I'm also not considered about the last edge case, if sprouts want us to parse opening hours, they need to conform to the spec.